cmd/guacone/cmd/patch.go

```diff
@@ -0,0 +1,397 @@
+	for _, id := range nodes {
+		node, err := model.Node(ctx, gqlClient, id)
+
+		if err != nil {
+			fmt.Printf("error: %s \n", err)
+		}
+
+		var pretty string
+		switch node := node.Node.(type) {
+		case *model.NodeNodePackage:
+			if bfsMap[id].Type == analysis.PackageName {
+				pretty = makePkgPretty(*node, false)
+			} else {
+				pretty = makePkgPretty(*node, true)
+			}
+		case *model.NodeNodeSource:
+			pretty = makeSrcPretty(*node)
+		case *model.NodeNodeArtifact:
+			pretty = makeArtifactPretty(*node)
+		}
+
+		fmt.Printf("%s: %s\n", id, pretty)
+
+		if bfsMap[id].PointOfContact.Email != "" {
+			poc = append(poc, id)
+		}
+	}
+	return poc
+}
+
+func makePOCPretty(poc model.AllPointOfContact) string {
+	return fmt.Sprintf("id- %s | email- %s | info- %s | collector- %s | justification- %s | origin- %s | since- %s", poc.Id, poc.Email, poc.Info, poc.Collector, poc.Justification, poc.Origin, poc.Since)
+}
+
+func makePkgPretty(pkg model.NodeNodePackage, version bool) string {
+	pkgString := fmt.Sprintf("pkg:%s/%s/%s", pkg.Type, pkg.Namespaces[0].Namespace, pkg.Namespaces[0].Names[0].Name)
+
+	if version {
+		return fmt.Sprintf("%s@%s", pkgString, pkg.Namespaces[0].Names[0].Versions[0].Version)
+	}
+
+	return pkgString
+}
+
+func makeSrcPretty(src model.NodeNodeSource) string {
+	return fmt.Sprintf("src:%s/%s/%s", src.Type, src.Namespaces[0].Namespace, src.Namespaces[0].Names[0].Name)
+}
+
+func makeArtifactPretty(artifact model.NodeNodeArtifact) string {
+	return fmt.Sprintf("artifact: algorithm-%s | digest:%s", artifact.Algorithm, artifact.Digest)
+}
+
+func getPkgID(ctx context.Context, gqlClient graphql.Client, purl string) (string, error) {
+	pkgInput, err := helpers.PurlToPkg(purl)
+
+	if err != nil {
+		return "", err
+	}
+
+	var pkgFilter model.PkgSpec
+	version := false
+
+	if pkgInput.Version != nil && *pkgInput.Version != "" {
+		pkgFilter = model.PkgSpec{
+			Type:      &pkgInput.Type,
+			Namespace: pkgInput.Namespace,
+			Name:      &pkgInput.Name,
+			Version:   pkgInput.Version,
+		}
+		version = true
+	} else {
+		pkgFilter = model.PkgSpec{
+			Type:      &pkgInput.Type,
+			Namespace: pkgInput.Namespace,
+			Name:      &pkgInput.Name,
+		}
+	}
+
+	pkgResponse, err := model.Packages(ctx, gqlClient, pkgFilter)
+
+	if err != nil || len(pkgResponse.Packages) == 0 {
+		return "", fmt.Errorf("error finding package with given purl: %s", purl)
+	}
+
+	if version {
+		return pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id, nil
+	}
+	return pkgResponse.Packages[0].Namespaces[0].Names[0].Id, nil
+}
+
+func validateQueryPatchFlags(graphqlEndpoint, startPurl string, stopPurl string, depth int, sampleData bool, args []string) (queryPatchOptions, error) {
+	var opts queryPatchOptions
+	opts.graphqlEndpoint = graphqlEndpoint
+	opts.startPurl = startPurl
+	opts.stopPurl = stopPurl
+	opts.depth = depth
+	opts.sampleData = sampleData
+
+	if len(opts.startPurl) > 0 && strings.Index(opts.startPurl, "pkg:") != 0 {
```

Comment:
move this check-up but also should be using `PurlToPkg`, if error its not valid: https://github.com/guacsec/guac/blob/main/pkg/assembler/helpers/purl.go